doc/ExpertOptions.md

```diff
@@ -685,16 +612,87 @@ The [WDL Spec](https://github.com/openwdl/wdl/blob/master/versions/1.0/SPEC.md#p
+streaming instead. The Unix `cat`, `wc`, and `head` commands are of
+this nature. To specify that a file is to be streamed, mark it as such
+in the `parameter_meta` section. For example:
+
+```wdl
+task head {
+    File in_file
+    Int num_lines
+
+    parameter_meta {
+        in_file : "stream"
+    }
+    command {
+        head -n ${num_lines} ${in_file}
+    }
+    output {
+        String result = read_string(stdout())
+    }
+}
+```
+
+File streaming is an optimization, and there are limiting rules to its
+correct usage. The file must be accessed only once, in sequential
+order, from the beginning. It need not be read to the end. If the task
+does not keep this contract, it could fail in unexpected ways.
+
+Some tasks have empty command sections. For example, the `fileSize`
+task (below) calculates the size of a file, but does not need to
+download it.  In such cases, the input files are downloaded lazily,
+only if their data is accessed.
+
+```wdl
+task fileSize {
+    File in_file
+
+    command {}
+    output {
+        Float num_bytes = size(in_file)
+    }
+}
+```
 
-There are several parameters affecting the runtime environment that can be specified in the dxapp.json file:
+## Task Runtime and Hints
 
-* `executionPolicy`: Specifies when to try to automatically restart failed jobs, and how many times
-* `timeoutPolicy`: Specifies the maximum amount of time the job can run
+### Instance type
+A task declaration has a runtime section where [native WDL runtime attributes(https://github.com/openwdl/wdl/blob/main/versions/1.1/SPEC.md#mandatory-runtime-attributes)] can be specified to customize its execution environment on the DNAnexus platform. Based on these attributes, an instance type is chosen by
+the compiler. If you wish to choose an instance type from the
+DNAnexus [native](https://documentation.dnanexus.com/developer/api/running-analyses/instance-types)
+list, this can be done by specifying the `dx_instance_type` key
+instead. For example:
+
+```
+runtime {
+   dx_instance_type: "mem1_ssd2_x4"
+}
+```
+
+If you want an instance that has a GPU chipset, set the `gpu` attribute to true. For example:
+```
+runtime {
+   memory: "4 GB"
+   cpu : 4
+   gpu : true
+}
+```
+CWL provides a different set of runtime attributes used in the ResourceRequirement, and an DNAnexus instance type will be chosen accordingly. Recognized attributes and its mapping to WDL is listed [here](CWL_v1.2.0_to_WDL_v1.md###requirements)
+
+### Addtional DNAnexus-specific runtime settings
+
+There are several parameters that also affect the runtime environment specified in the dxapp.json file https://documentation.dnanexus.com/developer/api/running-analyses/io-and-run-specifications#run-specification:
+* `runSpec.executionPolicy`: Specifies when to try to automatically restart failed jobs, and how many times
+* `runSpec.timeoutPolicy`: Specifies the maximum amount of time the job can run
 * `access`: Specifies which resources the applet can access
```

Comment:
Here I added the "runSpec." to note that these attrs need to be speciified under the key `runSpec` in extras,json
https://github.com/dnanexus/dxCompiler/blob/140a1c6f3e71ad6fd2072bbf794321daa5f851a3/doc/ExpertOptions.md#L1114
However `runSpec` and `access` and `ignoreReuse` are keys on the same level, so access should not be put under `runSpec`
https://documentation.dnanexus.com/developer/apps/app-metadata
But our current implementation of both `dx_access` in wdl file and `access` in extras.json are regarding access as a runtime attributes.
https://github.com/dnanexus/dxCompiler/blob/b0b0dab11eb203e71975eccb9fe25db0192a5400/compiler/src/main/scala/dx/translator/wdl/RuntimeTranslator.scala#L39
https://github.com/dnanexus/dxCompiler/blob/dd74c091f2ce2286c947dc334eb996fbdbc065b9/compiler/src/main/scala/dx/translator/Extras.scala#L449

Reply:
@commandlinegirl I think it is something to be fixed but not in this PR :(

Reply:
That's all right, even better. Thanks!